### PR TITLE
Filtering 3+ (or 2, in some circumstances) fields does not work

### PR DESCRIPTION
**Pull request by aaronbauman**
### Overview
This pull request demonstrates unexpected behavior when using 3 filters.

Expected:
- 3 filters get applied to results

Observed:
- Only the 1st filter (from left to right) works as expected.
- The 2nd and 3rd filters get mangled.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| Has tests?    | yes
| BC breaks?    | no     
| Deprecations? | no 

### Summary
Right now, this PR is only a test to demonstrate the behavior.

### Description
Still todo: fix the bug.

### CODE REVIEW

File: src/LogicalOpFactory.php [hunk 1]
```diff
@@ -92,13 +92,15 @@ protected function splitByLogicOp($expression)
             return [ [$expression, '', $expression] ];
         }
         $exprSet = [];
-        $i = 0;
-        foreach ($matches as $opWithOffset) {
-            list($op, $offset) = $opWithOffset[0];
```
Comment (aaronbauman) [inline src/LogicalOpFactory.php:97]: If there is any logic operator, there are always exactly 2 entries in `$matches`, so `$opWithOffset[0]` is always the same. We must iterate over `$matches[0]` (or `$matches[1]`) to find each individual expression.

File: src/LogicalOpFactory.php [hunk 2]
```diff
@@ -92,13 +92,15 @@ protected function splitByLogicOp($expression)
...
-            $expr = substr($expression, $i, $i + $offset);
```
Comment (aaronbauman) [inline src/LogicalOpFactory.php:98]: `$i + $offset` works only on the first iteration, when `$i == 0`.

It's only by coincidence that the logic evaluation works for 2 expressions, when the value of `$i+$offset` happens to be longer than total length of `$expression`. Instead, the way to capture a subsequent expression is `$offset - $i`.

File: src/LogicalOpFactory.php [hunk 3]
```diff
@@ -92,13 +92,15 @@ protected function splitByLogicOp($expression)
...
-            $exprSet[] = [ "$op$expr", $op, $expr, ];
+        $i = $offset = 0;
+        foreach ($matches[0] as $opWithOffset) {
+          list($op, $offset) = $opWithOffset;
+          $expr = substr($expression, $i, $offset - $i);
+          $i = $i + $offset + strlen($op);
+          $exprSet[] = [ "$op$expr", $op, $expr, ];
         }
+        $expr = substr($expression, $offset + strlen($op));
+        $exprSet[] = [ "$op$expr", $op, $expr];
```
Comment (aaronbauman) [inline src/LogicalOpFactory.php:104]: The `foreach` loop captures only N-1 expressions, so we have to grab one more.

**Comment by greg-1-anderson** [comment]
This looks good; I'm just wondering why the tests are not running.

**Comment by greg-1-anderson** [comment]
Ran tests in #47.

**Comment by aaronbauman** [comment]
> This looks good; I'm just wondering why the tests are not running.

iirc there was a message that I wasn't approved to trigger tests?
Or maybe because it was my first PR on this repo?
don't remember exactly, and it's not showing up anymore since the PR is closed.

**Comment by greg-1-anderson** [comment]
I've tried to minimize the GitHub settings to be as permissive as possible, but sometimes it's still a mystery. Maybe future PRs will run tests.

**Pull request merged**